src/pytest_inmanta_lsm/load_generator.py

```diff
@@ -0,0 +1,218 @@
+"""
+    Pytest Inmanta LSM
+
+    :copyright: 2024 Inmanta
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+
+import asyncio
+import datetime
+import functools
+import logging
+import threading
+import types
+import typing
+
+from pytest_inmanta_lsm import remote_orchestrator
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LoadException(Exception):  # noqa: N818
+    """A specific exception for the LoadGenerator: this means that the LoadGenerator thread has to stop running."""
+
+    pass
+
+
+class LoadGenerator:
+    """
+    This class helps to generate additional load on a real remote orchestrator by simulating the load produced by the dashboard.
+    """
+
+    def __init__(
+        self,
+        remote_orchestrator: remote_orchestrator.RemoteOrchestrator,
+        service_entity_name: str,
+        logger: logging.Logger = LOGGER,
+        sleep_time: float = 0.5,
+    ):
+        """
+        :param remote_orchestrator: The orchestrator, to which requests must be made
+        :param service_entity_name: The name of the service entity
+        :param logger: The logger to use
+        :param sleep_time: The time to wait between the different requests
+        """
+        self.remote_orchestrator = remote_orchestrator
+        self.service_entity_name = service_entity_name
+        self.sleep_time = sleep_time
+        self.logger = logger
+        self.running = True
+        self._thread: typing.Optional[threading.Thread] = None
+        self.exception: typing.Optional[Exception] = None
+
+    def __enter__(self) -> None:
+        self.logger.debug("Creating new Thread")
+        self._thread = threading.Thread(target=self.between_callback, daemon=True, name="Thread-LG")
+        self.logger.debug("Starting %s", self._thread.name)
+        self._thread.start()
+
+    def __exit__(
+        self,
+        exc_type: typing.Optional[typing.Type[BaseException]],
+        exc: typing.Optional[BaseException],
+        traceback: typing.Optional[types.TracebackType],
+    ) -> None:
+        self.running = False
+        self.logger.debug("%s should stop", self.thread.name)
+        self.logger.debug("Stopping %s", self.thread.name)
+        self.thread.join(self.remote_orchestrator.client.timeout)
+        self.logger.debug("%s has been stopped", self.thread.name)
+
+        if self.exception is not None:
+            raise self.exception
+
+    @property
+    def thread(self) -> threading.Thread:
+        if self._thread is None:
+            raise RuntimeError("The thread is not initialized, this instance should be used in a context!")
+
+        return self._thread
+
+    def between_callback(self) -> None:
+        """
+        Will create the load in an async manner. This method will also save any exception that could have occurred in the
+        thread. This mechanism allows to raise an exception once the thread has joined, see the `__exit__` method.
+        """
+        loop = asyncio.new_event_loop()
```

Comment:
Why combine asyncio with thread? i.e. is there a specific idea behind this?

Reply:
We rely on a thread to not block the main thread from doing anything other than running the test. And we rely on async because we use the `request` method of the remote orchestrator, which is async